include/net/http.h

```diff
@@ -277,13 +382,16 @@ struct http_client_request {
  * @brief Generic function to send a HTTP request to the network. Normally
  * applications would not need to use this function.
  *
+ * @param ctx HTTP client context.
  * @param net_ctx Network context.
  * @param req HTTP request to perform.
  * @param timeout Timeout when doing net_buf allocations.
  *
  * @return Return 0 if ok, and <0 if error.
  */
-int http_request(struct net_context *net_ctx, struct http_client_request *req,
+int http_request(struct http_client_ctx *ctx,
```

Comment:
Would it make sense to make net_contex* be part of http_client_ctx and not bother user to pass both?